src/main/java/couse_project/Game.java

```diff
@@ -0,0 +1,34 @@
+package couse_project;
+
+import java.io.IOException;
+
+public class Game {
```

Comment:
Good! You introduced some abstractions, showed knowledge of Java Core and Java libraries. The code in general looks good, but there are a few cases where I'm not sure about the correct behavior. Also, your project lacks abstract classes/interfaces, and some architecture (the simplest is service/model). The interface is cool, I like it. playing is quite comfortable) I also appreciate automatic ship placement and description, it really helps to test.

Approved!